Problems compiling HTTP_OTA on esp8266

Hello, i am really happy about this function, it is really what I was looking for, the problems is when i am going to compile, i receive this error:

Compiling .pioenvs/00-esp8285-batt-test/src/esphome/components/script/script.cpp.o
src/esphome/components/ota_http/ota_http.cpp: In function 'bool esphome::ota_http::http_connect(HTTPClient*, std::string)':
src/esphome/components/ota_http/ota_http.cpp:63:23: error: 'const class String' has no member named 'compare'; did you mean 'compareTo'?
   63 |   bool secure_ = url_.compare(0, 6, "https:") == 0;
      |                       ^~~~~~~
      |                       compareTo
src/esphome/components/ota_http/ota_http.cpp:68:28: error: 'using element_type = class WiFiClient' {aka 'class WiFiClient'} has no member named 'setInsecure'
   68 |       wifi_client_secure_->setInsecure();
      |                            ^~~~~~~~~~~
src/esphome/components/ota_http/ota_http.cpp:69:28: error: 'using element_type = class WiFiClient' {aka 'class WiFiClient'} has no member named 'setBufferSizes'
   69 |       wifi_client_secure_->setBufferSizes(512, 512);
      |                            ^~~~~~~~~~~~~~
src/esphome/components/ota_http/ota_http.cpp:86:18: error: 'client' was not declared in this scope; did you mean 'client_'?
   86 |   begin_status = client->begin(*wifi_client_, url_);
      |                  ^~~~~~
      |                  client_
Compiling .pioenvs/00-esp8285-batt-test/src/esphome/components/sensor/automation.cpp.o
*** [.pioenvs/00-esp8285-batt-test/src/esphome/components/ota_http/ota_http.cpp.o] Error 1
========================= [FAILED] Took 69.60 seconds =========================


Hope there is a solution, i am using an ESP8285 and this function is perfect to enable autoupdate from my own server, when the ESP is in another house and i do not want to open a port on the modem.
To test the option i copy your configuration and changed just the url

Thanks for this good option !!!!!!!!!!!!!!!!!!

I have problems to compile:

Compiling .pioenvs/00-esp8285-batt-test/src/esphome/components/sensor/filter.cpp.o
Compiling .pioenvs/00-esp8285-batt-test/src/esphome/components/sensor/sensor.cpp.o
src/esphome/components/ota_http/ota_http.cpp: In member function 'void esphome::ota_http::OtaHttpComponent::set_url(std::string)':
src/esphome/components/ota_http/ota_http.cpp:33:30: error: 'std::string' {aka 'class std::__cxx11::basic_string<char>'} has no member named 'startsWith'
   33 |   this->secure_ = this->url_.startsWith("https:") == 0;
      |                              ^~~~~~~~~~
Compiling .pioenvs/00-esp8285-batt-test/src/esphome/components/shutdown/switch/shutdown_switch.cpp.o
Compiling .pioenvs/00-esp8285-batt-test/src/esphome/components/sntp/sntp_component.cpp.o
*** [.pioenvs/00-esp8285-batt-test/src/esphome/components/ota_http/ota_http.cpp.o] Error 1
========================= [FAILED] Took 24.56 seconds =========================

At the moment i am using HTTP

--- a/components/ota_http/README.md
+++ b/components/ota_http/README.md
@@ -19,7 +19,6 @@ button:
       then:
         - ota_http.flash:
             url: http://example.com/firmware.bin
-            verify_ssl: false
         - logger.log: "This message should be not displayed(reboot)"
 ```
 

--- a/components/ota_http/ota_http.cpp
+++ b/components/ota_http/ota_http.cpp
@@ -109,8 +109,8 @@ void OtaHttpComponent::flash() {
   uint32_t update_start_time = millis();
   uint32_t start_time;
   uint32_t duration;
-  int http_code;
-  const char *headerKeys[] = {"Content-Length", "Content-Type"};
+  __attribute__(( aligned(4))) int http_code;
+  __attribute__(( aligned(4))) const char *headerKeys[] = {"Content-Length", "Content-Type"};
   const size_t headerCount = sizeof(headerKeys) / sizeof(headerKeys[0]);
   const size_t chunk_size = 1024;  // HTTP_TCP_BUFFER_SIZE;
   size_t chunk_start = 0;

--- a/components/ota_http/ota_http.cpp
+++ b/components/ota_http/ota_http.cpp
@@ -30,7 +30,7 @@ void OtaHttpComponent::dump_config() {
 
 void OtaHttpComponent::set_url(std::string url) {
   this->url_ = std::move(url);
-  this->secure_ = this->url_.compare(0, 6, "https:") == 0;
+  this->secure_ = this->url_.startsWith("https:") == 0;
 }
 
 std::unique_ptr<ota::OTABackend> make_ota_backend() {
@@ -88,9 +88,12 @@ bool http_connect(HTTPClient *client_, std::string url) {
 
   if (!begin_status) {
     ESP_LOGW(TAG, "Unable to make http connection");
+  } else {
+    ESP_LOGD(TAG, "Connected successfully.");
   }
 
   client_->setReuse(true);
+  ESP_LOGD(TAG, "http client setReuse.");
 
   return begin_status;
 }
@@ -141,14 +144,17 @@ void OtaHttpComponent::flash() {
   // we will compute md5 on the fly
   // TODO: better security if fetched from the http server
   md5_receive.init();
+  ESP_LOGD(TAG, "md5sum from received data initialized.");
 
   // returned needed headers must be collected before the requests
   client_.collectHeaders(headerKeys, headerCount);
+  ESP_LOGD(TAG, "http headers collected.");
 
   // http GET chunk
   start_time = millis();
   http_code = client_.GET();
   duration = millis() - start_time;
+  ESP_LOGD(TAG, "http GET finished.");
 
   if (http_code >= 310) {
     ESP_LOGW(TAG, "HTTP Request failed; URL: %s; Error: %s (%d); Duration: %u ms", url_.c_str(),

--- a/components/ota_http/ota_http.cpp
+++ b/components/ota_http/ota_http.cpp
@@ -60,9 +60,9 @@ bool http_connect(HTTPClient *client_, std::string url) {
   // ESP8266 code not tested
   std::shared_ptr<WiFiClient> wifi_client_;
 #ifdef USE_HTTP_REQUEST_ESP8266_HTTPS
-  bool secure_ = url_.compare(0, 6, "https:") == 0;
+  bool secure_ = url_.startsWith("https:") == 0;
   if (secure_) {
-    std::shared_ptr<WiFiClient> wifi_client_secure_;
+    std::shared_ptr<WiFiClientSecure> wifi_client_secure_;
     if (wifi_client_secure_ == nullptr) {
       wifi_client_secure_ = std::make_shared<BearSSL::WiFiClientSecure>();
       wifi_client_secure_->setInsecure();
@@ -83,7 +83,7 @@ bool http_connect(HTTPClient *client_, std::string url) {
   begin_status = client_->begin(url_);
 #elif defined(USE_ESP8266)
   // ESP8266 code not tested!
-  begin_status = client->begin(*wifi_client_, url_);
+  begin_status = client_->begin(*wifi_client_, url_);
 #endif
 
   if (!begin_status) {
@@ -107,7 +107,6 @@ void OtaHttpComponent::flash() {
   int http_code;
   const char *headerKeys[] = {"Content-Length", "Content-Type"};
   const size_t headerCount = sizeof(headerKeys) / sizeof(headerKeys[0]);
-  size_t total_size;
   const size_t chunk_size = 1024;  // HTTP_TCP_BUFFER_SIZE;
   size_t chunk_start = 0;
   size_t chunk_stop = chunk_size;
@@ -197,8 +196,8 @@ void OtaHttpComponent::flash() {
     update_started = true;
     error_code = backend->write(buf, bufsize);
     if (error_code != 0) {
-      esphome::ESP_LOGW(TAG, "Error code (%d) writing binary data to flash at offset %d/%d and size %s", error_code,
-                        chunk_start, total_size, body_lenght);
+      esphome::ESP_LOGW(TAG, "Error code (%d) writing binary data to flash at offset %d and size %s", error_code,
+                        chunk_start, body_lenght);
       goto error;  // NOLINT(cppcoreguidelines-avoid-goto)
     }
 

--- a/components/ota_http/ota_http.cpp
+++ b/components/ota_http/ota_http.cpp
@@ -30,7 +30,7 @@ void OtaHttpComponent::dump_config() {
 
 void OtaHttpComponent::set_url(std::string url) {
   this->url_ = std::move(url);
-  this->secure_ = this->url_.startsWith("https:") == 0;
+  this->secure_ = this->url_.rfind("https:", 0) == 0;
 }
 
 std::unique_ptr<ota::OTABackend> make_ota_backend() {

--- a/components/ota_http/ota_http.cpp
+++ b/components/ota_http/ota_http.cpp
@@ -50,7 +50,7 @@ std::unique_ptr<ota::OTABackend> make_ota_backend() {
   ESP_LOGD(TAG, "Using ArduinoRP2040OTABackend");
   return make_unique<ota::ArduinoRP2040OTABackend>();
 #endif  // USE_RP2040
-  esphome::ESP_LOGE(TAG, "No OTA backend!");
+  ESP_LOGE(TAG, "No OTA backend!");
 }
 
 struct Header {
@@ -156,19 +156,19 @@ void OtaHttpComponent::flash() {
   while (bytes_read != body_length) {
     size_t bufsize = std::min(chunk_size, body_length - bytes_read);
 
-    ESP_LOGVV(TAG, "going to %d bytes at %zu/%zu", bufsize, bytes_read, body_length);
+    // ESP_LOGVV(TAG, "going to %d bytes at %zu/%zu", bufsize, bytes_read, body_length);
 
-    ESP_LOGVV(TAG, "waiting for %zu bytes available..", bufsize);
+    // ESP_LOGVV(TAG, "waiting for %zu bytes available..", bufsize);
     while (stream.available() < bufsize) {
       // give other tasks a chance to run while waiting for some data:
-      ESP_LOGVV(TAG, "data available: %zu", stream.available());
+      // ESP_LOGVV(TAG, "data available: %zu", stream.available());
       yield();
       delay(10);
     }
-    ESP_LOGVV(TAG, "data available: %zu", stream.available());
+    // ESP_LOGVV(TAG, "data available: %zu", stream.available());
 
     stream.readBytes(buf, bufsize);
-    ESP_LOGVV(TAG, "buffer read");
+    // ESP_LOGVV(TAG, "buffer read");
     if (bytes_read == 0 and buf[0] != 0xE9) {
       // check magic byte
       ESP_LOGE(TAG, "Firmware magic byte 0xE9 a pos 0 failed! OTA aborted");
@@ -178,14 +178,14 @@ void OtaHttpComponent::flash() {
     buf[bufsize] = '\0';  // not fed to ota
 
     md5_receive.add(buf, bufsize);
-    ESP_LOGVV(TAG, "md5 added");
+    // ESP_LOGVV(TAG, "md5 added");
 
     update_started = true;
     error_code = backend->write(buf, bufsize);
-    ESP_LOGVV(TAG, "wrote to backend");
+    // ESP_LOGVV(TAG, "wrote to backend");
     if (error_code != 0) {
-      esphome::ESP_LOGW(TAG, "Error code (%d) writing binary data to flash at offset %d and size %s", error_code,
-                        chunk_start, body_length);
+      ESP_LOGW(TAG, "Error code (%d) writing binary data to flash at offset %d and size %s", error_code, chunk_start,
+               body_length);
       goto error;  // NOLINT(cppcoreguidelines-avoid-goto)
     }
 
@@ -193,7 +193,7 @@ void OtaHttpComponent::flash() {
     uint32_t now = millis();
     if ((now - last_progress > 1000) or (bytes_read == body_length)) {
       last_progress = now;
-      esphome::ESP_LOGD(TAG, "Progress: %0.1f%%", bytes_read * 100. / body_length);
+      ESP_LOGD(TAG, "Progress: %0.1f%%", bytes_read * 100. / body_length);
 
       // feed watchdog and give other tasks a chance to run
       esphome::App.feed_wdt();
@@ -202,12 +202,12 @@ void OtaHttpComponent::flash() {
 
   }  // while
 
-  esphome::ESP_LOGI(TAG, "Done in %.0f secs", (millis() - update_start_time) / 1000);
+  ESP_LOGI(TAG, "Done in %.0f secs", (millis() - update_start_time) / 1000);
 
   // send md5 to backend (backend will check that the flashed one has the same)
   md5_receive.calculate();
   md5_receive.get_hex(md5_receive_str);
-  esphome::ESP_LOGD(TAG, "md5sum recieved: %s (size %d)", md5_receive_str, bytes_read);
+  ESP_LOGD(TAG, "md5sum recieved: %s (size %d)", md5_receive_str, bytes_read);
   backend->set_update_md5(md5_receive_str);
 
   client_.end();
@@ -221,12 +221,12 @@ void OtaHttpComponent::flash() {
 
   error_code = backend->end();
   if (error_code != 0) {
-    esphome::ESP_LOGW(TAG, "Error ending OTA!, error_code: %d", error_code);
+    ESP_LOGW(TAG, "Error ending OTA!, error_code: %d", error_code);
     goto error;  // NOLINT(cppcoreguidelines-avoid-goto)
   }
 
   delay(10);
-  esphome::ESP_LOGI(TAG, "OTA update finished! Rebooting...");
+  ESP_LOGI(TAG, "OTA update finished! Rebooting...");
   delay(100);  // NOLINT
   esphome::App.safe_reboot();
   // new firmware flashed!

--- a/components/ota_http/ota_http.cpp
+++ b/components/ota_http/ota_http.cpp
@@ -60,10 +60,12 @@ bool http_connect(HTTPClient *client_, std::string url) {
   // ESP8266 code not tested
   std::shared_ptr<WiFiClient> wifi_client_;
 #ifdef USE_HTTP_REQUEST_ESP8266_HTTPS
-  bool secure_ = url_.startsWith("https:") == 0;
+  bool secure_ = url_.startsWith("https:");
   if (secure_) {
+    ESP_LOGD(TAG, "https connection");
     std::shared_ptr<WiFiClientSecure> wifi_client_secure_;
     if (wifi_client_secure_ == nullptr) {
+      ESP_LOGD(TAG, "Fallback to BearSSL::WiFiClientSecure>()");
       wifi_client_secure_ = std::make_shared<BearSSL::WiFiClientSecure>();
       wifi_client_secure_->setInsecure();
       wifi_client_secure_->setBufferSizes(512, 512);

--- a/components/ota_http/ota_http.cpp
+++ b/components/ota_http/ota_http.cpp
@@ -168,12 +168,6 @@ void OtaHttpComponent::flash() {
     // ESP_LOGVV(TAG, "data available: %zu", stream.available());
 
     stream.readBytes(buf, bufsize);
-    // ESP_LOGVV(TAG, "buffer read");
-    if (bytes_read == 0 and buf[0] != 0xE9) {
-      // check magic byte
-      ESP_LOGE(TAG, "Firmware magic byte 0xE9 a pos 0 failed! OTA aborted");
-      return;
-    }
     bytes_read += bufsize;
     buf[bufsize] = '\0';  // not fed to ota
 
@@ -233,7 +227,9 @@ void OtaHttpComponent::flash() {
 
 error:
   if (update_started) {
+    ESP_LOGE(TAG, "Aborted");
     backend->abort();
+    return;
   }
 }
 

--- a/components/ota_http/ota_http.cpp
+++ b/components/ota_http/ota_http.cpp
@@ -3,7 +3,7 @@
 // code adapted from
 // esphome/components/ota/ota_backend.cpp
 // and
-// /esphome/components/http_request
+// esphome/components/http_request
 #ifdef USE_ARDUINO
 
 #include "ota_http.h"
@@ -53,53 +53,6 @@ std::unique_ptr<ota::OTABackend> make_ota_backend() {
   esphome::ESP_LOGE(TAG, "No OTA backend!");
 }
 
-bool http_connect(HTTPClient *client_, std::string url) {
-  const String url_ = url.c_str();
-
-#ifdef USE_ESP8266
-  // ESP8266 code not tested
-  std::shared_ptr<WiFiClient> wifi_client_;
-#ifdef USE_HTTP_REQUEST_ESP8266_HTTPS
-  bool secure_ = url_.startsWith("https:");
-  if (secure_) {
-    ESP_LOGD(TAG, "https connection");
-    std::shared_ptr<WiFiClientSecure> wifi_client_secure_;
-    if (wifi_client_secure_ == nullptr) {
-      ESP_LOGD(TAG, "Fallback to BearSSL::WiFiClientSecure>()");
-      wifi_client_secure_ = std::make_shared<BearSSL::WiFiClientSecure>();
-      wifi_client_secure_->setInsecure();
-      wifi_client_secure_->setBufferSizes(512, 512);
-    }
-    wifi_client_ = wifi_client_secure_;
-  }
-#endif
-  if (wifi_client_ == nullptr) {
-    wifi_client_ = std::make_shared<WiFiClient>();
-  }
-#endif
-
-  ESP_LOGD(TAG, "Trying to connect to %s", url_.c_str());
-
-  bool begin_status = false;
-#if defined(USE_ESP32)
-  begin_status = client_->begin(url_);
-#elif defined(USE_ESP8266)
-  // ESP8266 code not tested!
-  begin_status = client_->begin(*wifi_client_, url_);
-#endif
-
-  if (!begin_status) {
-    ESP_LOGW(TAG, "Unable to make http connection");
-  } else {
-    ESP_LOGD(TAG, "Connected successfully.");
-  }
-
-  client_->setReuse(true);
-  ESP_LOGD(TAG, "http client setReuse.");
-
-  return begin_status;
-}
-
 struct Header {
   const char *name;
   const char *value;
@@ -109,8 +62,8 @@ void OtaHttpComponent::flash() {
   uint32_t update_start_time = millis();
   uint32_t start_time;
   uint32_t duration;
-  __attribute__(( aligned(4))) int http_code;
-  __attribute__(( aligned(4))) const char *headerKeys[] = {"Content-Length", "Content-Type"};
+  int http_code;
+  const char *headerKeys[] = {"Content-Length", "Content-Type"};
   const size_t headerCount = sizeof(headerKeys) / sizeof(headerKeys[0]);
   const size_t chunk_size = 1024;  // HTTP_TCP_BUFFER_SIZE;
   size_t chunk_start = 0;
@@ -121,13 +74,12 @@ void OtaHttpComponent::flash() {
   bool update_started = false;
   int error_code = 0;
   uint32_t last_progress = 0;
-  size_t body_lenght;
+  size_t body_length;
   size_t bytes_read = 0;
   esphome::md5::MD5Digest md5_receive;
   char *md5_receive_str = new char[33];
   HTTPClient client_{};
   WiFiClient stream;
-  // Stream stream;
   std::unique_ptr<ota::OTABackend> backend;
 
   if (!network::is_connected()) {
@@ -135,28 +87,48 @@ void OtaHttpComponent::flash() {
     return;
   }
 
-  // connect to http server using this->url_
-  bool status = http_connect(&client_, this->url_);
-  if (!status) {
-    client_.end();
-    ESP_LOGW(TAG, "HTTP Request failed at the begin phase. Please check the configuration");
-    return;
+#ifdef USE_ESP8266
+#ifdef USE_HTTP_REQUEST_ESP8266_HTTPS
+  if (this->secure_) {
+    ESP_LOGE(TAG, "https connection not handled!");
   }
+#endif  // USE_HTTP_REQUEST_ESP8266_HTTPS
+#endif  // USE_ESP8266
+
+  ESP_LOGD(TAG, "Trying to connect to %s", this->url_.c_str());
+
+  bool status = false;
+#if defined(USE_ESP32)
+  status = client_.begin(this->url_.c_str());
+#elif defined(USE_ESP8266)
+  status = client_.begin(stream, this->url_.c_str());
+#endif
+
+  if (!status) {
+    ESP_LOGE(TAG, "Unable to make http connection");
+    client_.end();
+    return;
+  } else {
+    ESP_LOGV(TAG, "http begin successfull.");
+  }
+
+  client_.setReuse(true);
+  ESP_LOGVV(TAG, "http client setReuse.");
 
   // we will compute md5 on the fly
   // TODO: better security if fetched from the http server
   md5_receive.init();
-  ESP_LOGD(TAG, "md5sum from received data initialized.");
+  ESP_LOGV(TAG, "md5sum from received data initialized.");
 
   // returned needed headers must be collected before the requests
   client_.collectHeaders(headerKeys, headerCount);
-  ESP_LOGD(TAG, "http headers collected.");
+  ESP_LOGV(TAG, "http headers collected.");
 
-  // http GET chunk
+  // http GET
   start_time = millis();
   http_code = client_.GET();
   duration = millis() - start_time;
-  ESP_LOGD(TAG, "http GET finished.");
+  ESP_LOGV(TAG, "http GET finished.");
 
   if (http_code >= 310) {
     ESP_LOGW(TAG, "HTTP Request failed; URL: %s; Error: %s (%d); Duration: %u ms", url_.c_str(),
@@ -164,31 +136,39 @@ void OtaHttpComponent::flash() {
     return;
   }
 
-  body_lenght = client_.getSize();
+  body_length = client_.getSize();
+  ESP_LOGD(TAG, "firmware is %d bytes length.", body_length);
 
   // flash memory backend
   backend = make_ota_backend();
-  error_code = backend->begin(body_lenght);
+
+  error_code = backend->begin(body_length);
   if (error_code != 0) {
     ESP_LOGW(TAG, "backend->begin error: %d", error_code);
     goto error;  // NOLINT(cppcoreguidelines-avoid-goto)
   }
 
-  // the data is read chunked
+#ifdef ESP32
   stream = client_.getStream();
+  ESP_LOGVV(TAG, "Got esp32 stream");
+#endif
 
-  while (bytes_read != body_lenght) {
-    size_t bufsize = std::min(chunk_size, body_lenght - bytes_read);
+  while (bytes_read != body_length) {
+    size_t bufsize = std::min(chunk_size, body_length - bytes_read);
 
-    // ESP_LOGD(TAG, "going to read %d/%d", bufsize, body_lenght);
-    //  binary read
+    ESP_LOGVV(TAG, "going to %d bytes at %zu/%zu", bufsize, bytes_read, body_length);
 
+    ESP_LOGVV(TAG, "waiting for %zu bytes available..", bufsize);
     while (stream.available() < bufsize) {
       // give other tasks a chance to run while waiting for some data:
+      ESP_LOGVV(TAG, "data available: %zu", stream.available());
       yield();
-      delay(1);
+      delay(10);
     }
+    ESP_LOGVV(TAG, "data available: %zu", stream.available());
+
     stream.readBytes(buf, bufsize);
+    ESP_LOGVV(TAG, "buffer read");
     if (bytes_read == 0 and buf[0] != 0xE9) {
       // check magic byte
       ESP_LOGE(TAG, "Firmware magic byte 0xE9 a pos 0 failed! OTA aborted");
@@ -196,24 +176,24 @@ void OtaHttpComponent::flash() {
     }
     bytes_read += bufsize;
     buf[bufsize] = '\0';  // not fed to ota
-    // ESP_LOGD(TAG, "buf: -%s- read %d/%d", (char *)buf, bytes_read, body_lenght); // string only!
-    // ESP_LOGD(TAG, "read %d/%d", bytes_read, body_lenght);
 
     md5_receive.add(buf, bufsize);
+    ESP_LOGVV(TAG, "md5 added");
 
     update_started = true;
     error_code = backend->write(buf, bufsize);
+    ESP_LOGVV(TAG, "wrote to backend");
     if (error_code != 0) {
       esphome::ESP_LOGW(TAG, "Error code (%d) writing binary data to flash at offset %d and size %s", error_code,
-                        chunk_start, body_lenght);
+                        chunk_start, body_length);
       goto error;  // NOLINT(cppcoreguidelines-avoid-goto)
     }
 
     // show progress, and feed watch dog
     uint32_t now = millis();
-    if ((now - last_progress > 1000) or (bytes_read == body_lenght)) {
+    if ((now - last_progress > 1000) or (bytes_read == body_length)) {
       last_progress = now;
-      esphome::ESP_LOGD(TAG, "Progress: %0.1f%%", bytes_read * 100. / body_lenght);
+      esphome::ESP_LOGD(TAG, "Progress: %0.1f%%", bytes_read * 100. / body_length);
 
       // feed watchdog and give other tasks a chance to run
       esphome::App.feed_wdt();
@@ -222,7 +202,7 @@ void OtaHttpComponent::flash() {
 
   }  // while
 
-  esphome::ESP_LOGD(TAG, "Done in %.0f secs", (millis() - update_start_time) / 1000);
+  esphome::ESP_LOGI(TAG, "Done in %.0f secs", (millis() - update_start_time) / 1000);
 
   // send md5 to backend (backend will check that the flashed one has the same)
   md5_receive.calculate();

--- a/components/ota_http/ota_http.h
+++ b/components/ota_http/ota_http.h
@@ -10,7 +10,6 @@
 #include <utility>
 #include <string>
 
-
 #ifdef USE_ESP32
 #include <HTTPClient.h>
 #endif
@@ -28,48 +27,44 @@ namespace ota_http {
 // in HTTPClient.h : #define HTTP_TCP_BUFFER_SIZE (1460)
 
 class OtaHttpComponent : public Component {
-    public:
-        void dump_config() override;
-        float get_setup_priority() const override { return setup_priority::AFTER_WIFI; }
-        void set_url(std::string url);
-        void set_timeout(uint64_t timeout) { this->timeout_ = timeout; }
-        // void flash(const std::vector<HttpRequestResponseTrigger *> &response_triggers);
-        void flash();
-    
-    protected:
-        
-        std::string url_;
-        bool secure_;
-        uint64_t timeout_{1000*60*10};  // must match CONF_TIMEOUT in __init__.py
+ public:
+  void dump_config() override;
+  float get_setup_priority() const override { return setup_priority::AFTER_WIFI; }
+  void set_url(std::string url);
+  void set_timeout(uint64_t timeout) { this->timeout_ = timeout; }
+  // void flash(const std::vector<HttpRequestResponseTrigger *> &response_triggers);
+  void flash();
+
+ protected:
+  std::string url_;
+  bool secure_;
+  uint64_t timeout_{1000 * 60 * 10};  // must match CONF_TIMEOUT in __init__.py
 };
 
 template<typename... Ts> class OtaHttpFlashAction : public Action<Ts...> {
-    public:
-        OtaHttpFlashAction(OtaHttpComponent *parent) : parent_(parent) {}
-        TEMPLATABLE_VALUE(std::string, url)
-        TEMPLATABLE_VALUE(uint16_t, timeout)
+ public:
+  OtaHttpFlashAction(OtaHttpComponent *parent) : parent_(parent) {}
+  TEMPLATABLE_VALUE(std::string, url)
+  TEMPLATABLE_VALUE(uint16_t, timeout)
 
-        // void register_response_trigger(HttpRequestResponseTrigger *trigger) { this->response_triggers_.push_back(trigger); }
+  // void register_response_trigger(HttpRequestResponseTrigger *trigger) { this->response_triggers_.push_back(trigger);
+  // }
 
-        void play(Ts... x) override {
-            this->parent_->set_url(this->url_.value(x...));
-            if (this->timeout_.has_value()) {
-                this->parent_->set_timeout(this->timeout_.value(x...));
-            }
-            // this->parent_->flash(this->response_triggers_);
-            this->parent_->flash();
-        }
-    
-    protected:
-        OtaHttpComponent *parent_;
-        // std::vector<HttpRequestResponseTrigger *> response_triggers_;
+  void play(Ts... x) override {
+    this->parent_->set_url(this->url_.value(x...));
+    if (this->timeout_.has_value()) {
+      this->parent_->set_timeout(this->timeout_.value(x...));
+    }
+    // this->parent_->flash(this->response_triggers_);
+    this->parent_->flash();
+  }
+
+ protected:
+  OtaHttpComponent *parent_;
+  // std::vector<HttpRequestResponseTrigger *> response_triggers_;
 };
 
-
-                                                        
-
-
-}  // namespace http_request
+}  // namespace ota_http
 }  // namespace esphome
 
 #endif  // USE_ARDUINO

